bluepy/btle.py

```diff
@@ -455,9 +455,9 @@ def _connect(self, addr, addrType=ADDR_TYPE_PUBLIC, iface=None, timeout=None):
         if rsp is None:
             raise BTLEDisconnectError("Timed out while trying to connect to peripheral %s, addr type: %s" %
                                       (addr, addrType), rsp)
-        while rsp['state'][0] == 'tryconn':
+        while rsp and rsp['state'][0] == 'tryconn':
             rsp = self._getResp('stat', timeout)
-        if rsp['state'][0] != 'conn':
```

Comment:
if `resp is None` that means a `timeout` happend. So a simular exception as on line 456 should be raised after `self._stopHelper()` is called.